libs/coroutines/include/hpx/coroutines/detail/coroutine_impl.hpp

```diff
@@ -113,15 +114,19 @@ namespace hpx { namespace threads { namespace coroutines { namespace detail {
 
         void reset()
         {
-            this->reset_stack();
-            m_result = result_type(terminated, invalid_thread_id);
```

Comment:
I've removed resetting `m_result` completely here as this was anyway overwriting the result set with `bind_result`. As far as I can tell we don't rely on this being reset here. This lets us keep the assertion in `bind_result` that says that we can't bind a result if the result is already `terminated`.

Reply:
This is fine as long as we end up resetting things in the end for each thread-reuse.